### ISSUE TITLE
Lower the default MacOS deployment target in the XCode project

### ISSUE THREAD

**Issue by Drabble** [maplibre/maplibre-rs#58]
The MacOS deployment target in the XCode project is set by default to 12.3. I may be wrong but from what I understand, a Mac on version 12.3 can run applications that target a lower version but it doesn't work the other way around. A Mac on version 11 can't run applications with a target version of 12.

I think we should target in our MacOS build the lowest version possible that supports the APIs we are using. I have tried running with a target of 10.9 (which is the lowest version proposed on XCode by default) and it seems to work alright.

This would make the configuration of the XCode IDE easier for new developers. The MacOS build would also support a wider range of devices.

**Comment by maxammann** [comment]
Yes that makes sense. I just went with the defaults feom xcode. Lets just use the lowest shown by xcode. We probably can go even lower.

I think we only depend (implicitly) on UIKit/AppKit, Metal and QuartzCore.
